### PR TITLE
Developer documentation

### PR DESCRIPTION
**Pull request by edcdavid**
This is a developer documentation for the PTP conformance tests. Also contains some minor files reorganization.
Doc file is at: https://github.com/openshift/ptp-operator/blob/dd57ea01c95fdde6341043a31ccbe046c3589beb/test/DEVELOPERS.md

### CODE REVIEW

File: test/DEVELOPERS.md [hunk 1]
```diff
@@ -0,0 +1,879 @@
...
+
+![overview_bc](doc/bcoverview.svg)
+
+### Dual NIC Boundary Clock
+Dual NIC BC are similar to boundary clocks except that their provide more master ports to synchronize more ordinary/boundary clocks. Network cards are usually limited to a maximum of 4 ports sharing a PTP Hardware clock. If more than 4 downstream clocks need to received synchronization 2 NIC can be configured together using the Dual NIC BC pattern. 
+
+A primary Boundary clock receives synchronization from a master clock and a secondary boundary clock also receives updates from a Master clock which synchronizes its PHC. However, the secondary BC does not run phc2sys to synchronize its PHC with the OS clock. If the Primary connection to the Grandmaster is lost, the secondary BC acts as the primary.
```
Comment (aneeshkp) [inline test/DEVELOPERS.md:25]: If the Primary connection to the Grandmaster is lost, the secondary BC acts as the primary. We don't run this as HA. so both will be synching differently downstream. if the primary is lost, The secondary won't sync primary PHC

File: test/DEVELOPERS.md [hunk 2]
```diff
@@ -0,0 +1,879 @@
...
+Steps:
+- configure ptp events in ptp-operator
+- configure min/max offset values in the ptpconfig using the ptpClockThreshold object:
+```
+    ptpClockThreshold:
+      holdOverTimeout: 5
+      maxOffsetThreshold: 100
+      minOffsetThreshold: -100
+```
+- instantiate a consumer sidecar and configure it to connect to the right ptp event service: `ptp-event-publisher-service-<nodename>`. where `<nodename>` is the nodename where the clock under test is configured.
```
Comment (aneeshkp) [inline test/DEVELOPERS.md:387]: .NAMESPACE.svc.cluster.local:9043

File: test/DEVELOPERS.md (hunk 1 above)
Comment (edcdavid) [reply test/DEVELOPERS.md]: fixed

File: test/DEVELOPERS.md (hunk 2 above)
Comment (edcdavid) [reply test/DEVELOPERS.md]: fixed

File: test/DEVELOPERS.md [hunk 3]
```diff
@@ -0,0 +1,879 @@
...
+      masterOnly 0
+      [ens7f1]
+      masterOnly 1
+      [ens7f2]
+      masterOnly 1
+  recommend:
+  - profile: "profile1"
+    priority: 0
+    match:
+    - nodeLabel: "node-role.kubernetes.io/worker"
```
[the remark this answers is not in the document]
Comment (edcdavid) [reply test/DEVELOPERS.md]: fixed

**Review by aneeshkp — APPROVED** [review]

**Pull request merged**